db/migrate/20200407140854_rename_budget_activity_id_to_parent_activity_id.rb

```diff
@@ -0,0 +1,29 @@
+class RenameBudgetActivityIdToParentActivityId < ActiveRecord::Migration[6.0]
+  def up
+    add_column :budgets, :parent_activity_id, :uuid
+    add_index :budgets, :parent_activity_id
+
+    ActiveRecord::Base.transaction do
+      Budget.all.each do |budget|
+        budget.parent_activity = budget.activity
+        budget.save!
+      end
+    end
+
+    remove_column :budgets, :activity_id
+  end
```

Comment:
I wonder how this is going to play now that the database contraints have gone in ahead of this. Do we need to add_foreign_key to the new column?

Reply:
I think I've amended this correctly but it'd be good if you can scrutinise it for me to make sure I've got it right

Reply:
It looks like the foreign key constraint for the new field is ok. I wonder if we need to remove the old foreign key constraint? 

An idea could be to checkout develop along with setting the correct database state. Then checkout your branch and apply this migration. If that works up, and then on a rollback we're likely ok 👍